lib/g_searcher_web/controllers/report_controller.ex

```diff
@@ -32,10 +32,22 @@ defmodule GSearcherWeb.ReportController do
   def show(conn, %{"id" => report_id}) do
     user = conn.assigns.current_user
 
-    case Reports.get_by(%{id: report_id, user_id: user.id}) do
-      {:ok, report} ->
-        render(conn, "show.html", report: report)
-
+    with {:ok, report} <- Reports.get_by(%{id: report_id, user_id: user.id}),
+         {:ok, search_results} <- SearchResults.get_search_results_from_report(report_id),
+         search_results_with_index <- Enum.with_index(search_results, 1),
+         total_keyword_count <- Reports.total_report_keywords_count(report_id),
+         total_searched_keyword_count <- Reports.total_searched_report_keywords_count(report_id),
+         total_top_advertisers_count <- Reports.total_top_advertisers_count(report_id),
+         report_status <- Reports.report_status(report) do
```

Comment:
These could be outside with, I think? 🤔  Seems it will always returning a value. Can we put it directly in the render below?